streams/src/main/java/org/apache/kafka/streams/state/internals/QueryableStoreProvider.java

```diff
@@ -58,9 +58,21 @@ public QueryableStoreProvider(final List<StreamThreadStateStoreProvider> storePr
         }
         final List<T> allStores = new ArrayList<>();
         for (final StreamThreadStateStoreProvider storeProvider : storeProviders) {
-            allStores.addAll(storeProvider.stores(storeQueryParameters));
+            final List<T> stores = storeProvider.stores(storeQueryParameters);
+            if (stores != null && !stores.isEmpty()) {
+                allStores.addAll(stores);
+                if (storeQueryParameters.partition() != null) {
+                    break;
+                }
+            }
         }
         if (allStores.isEmpty()) {
+            if (storeQueryParameters.partition() != null) {
```

Comment:
We should extend `QueryableStoreProviderTest` for this case -- throwing this exception moves from `StreamThreadStateStoreProvider` to here and we should not just remove the test from `StreamThreadStateStoreProviderTest` but add a new one to `QueryableStoreProviderTest`, too.

Reply:
I extended StateStoreProviderStub functionality to add store by partition with default partition 0, + relevant tests in QueryableStoreProviderTest